File: haas_lib_bundles/python/libraries/photoresistor/photoresistor.py
from driver import ADC

class PHOTORESISTOR(object):

    def __init__(self, adcObj):
        self.adcObj = None
        if not isinstance(adcObj, ADC):
            raise ValueError("parameter is not an ADC object")

        self.adcObj = adcObj

    def getLightness(self):
        if self.adcObj is None:
            raise ValueError("invalid ADC object")

        value = self.adcObj.readVoltage()
        return value

File: components/amp/example-js/ESP32/examples/smart_public_lighting/photoresistor.js
class _PhotoResistor {
    constructor(adcObj) {
        this.adcInstance = adcObj;
    };

    getLightness() {
	var value = this.adcInstance.read();
	return value;
    };
}

export function PhotoResistor(adcObj) {
    return new _PhotoResistor(adcObj);
}
